stoservice/src/main/java/com/tutrit/stoservice/controller/UserController.java

```diff
@@ -14,8 +14,18 @@ public UserController(UserService userService) {
 
     @Override
     public void doCommand(Request request, Response response) {
-        userService.saveUser(ParseUser.parseCommand(request));
-        response.setResponse("new user has been saved");
+        if (commandName(request).equals("register")){
+            userService.saveUserWithDiscount(ParseUser.parseCommand(request));
+            response.setResponse("new user has been saved and got a 5% discount!");
+        } else {
+            userService.saveUser(ParseUser.parseCommand(request));
+            response.setResponse("new user has been saved");
+        }
+    }
+    public String commandName(Request request) {
+        String userInput = request.getCommand();
+        String[] parseMap = userInput.split("-");
+        return parseMap[0] = parseMap[0].strip();
```

Comment:
We already have method that retrieves the command from request.
You can remove this method and change to this one
`com.tutrit.stoservice.utils.GetCommand.getCommand`